lib/rubocop/cop/rspec/mixin/top_level_group.rb

```diff
@@ -28,7 +31,10 @@ def on_top_level_example_group(_node); end
 
         def on_top_level_group(_node); end
 
+        # @private
+        # @deprecated All callers of this method have been removed.
         def top_level_group?(node)
+          warn DEPRECATED_PRIVATE_METHOD_WARNING
```

Comment:
In RuboCop [they use](https://github.com/rubocop/rubocop/blob/87740810904f57499e1255d492b80b4bd83d3a66/lib/rubocop/cop/mixin/configurable_max.rb#L12-L14) `Rainbow` to make yellow warnings. I’m a bit hesitant to add it as a dependency to our gem, but I think we should use the `uplevel` keyword argument that they also use:
```suggestion
          warn DEPRECATED_PRIVATE_METHOD_WARNING, uplevel: 1
```
That also prints the line that called the method, plus the word `warning`.